sdk/core/src/error/http_error.rs

```diff
@@ -16,13 +17,15 @@ impl HttpError {
     pub async fn new(response: Response) -> Self {
         let status = response.status();
         let mut error_code = get_error_code_from_header(&response);
-        let headers = response
-            .headers()
-            .into_iter()
-            // TODO: the following will panic if a non-UTF8 header value is sent back
-            // We should not panic but instead handle this gracefully
-            .map(|(n, v)| (n.as_str().to_owned(), v.to_str().unwrap().to_owned()))
-            .collect();
+        let mut headers = HashMap::new();
+
+        for (name, value) in response.headers() {
+            // TODO: non-UTF8 header values will not be included
```

Comment:
I suppose if we cannot convert to a string, we could convert to bytes and then use `String::from_utf8_lossy`.

Reply:
addressed